### ISSUE TITLE
org.openqa.selenium.SessionNotCreatedException: Unable to create new service: ChromeDriverService

### ISSUE THREAD

**Issue by MFaizanChughtai** [SeleniumHQ/selenium#6392]
Starting ChromeDriver 2.41.578737 (49da6702b16031c40d63e5618de03a32ff6c197e) on port 14596
Only local connections are allowed.
Sep 11, 2018 12:39:56 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: OSS
Sep 11, 2018 12:39:57 PM org.openqa.selenium.remote.DesiredCapabilities chrome
INFO: Using `new ChromeOptions()` is preferred to `DesiredCapabilities.chrome()`
org.openqa.selenium.SessionNotCreatedException: Unable to create new service: ChromeDriverService
Build info: version: '3.9.1', revision: '63f7b50', time: '2018-02-07T22:42:28.403Z'
System info: host: 'WRIT-A05', ip: '172.16.7.127', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_111'
Driver info: driver.version: unknown
Command duration or timeout: 340 milliseconds
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:214)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:166)
	at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$new$0(JsonWireProtocolResponse.java:53)
	at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$getResponseFunction$2(JsonWireProtocolResponse.java:91)
	at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:123)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Unknown Source)
	at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:126)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:73)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:138)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:601)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:219)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:142)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:155)
	at ISCO.ISCO.invokeBrowser(ISCO.java:59)
	at ISCO.ISCO.main(ISCO.java:534)
Caused by: org.openqa.selenium.SessionNotCreatedException: Unable to create new service: ChromeDriverService
Build info: version: '3.9.1', revision: '63f7b50', time: '2018-02-07T22:42:28.403Z'
System info: host: 'WRIT-A05', ip: '172.16.7.127', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_111'
Driver info: driver.version: unknown
Build info: version: '3.9.1', revision: '63f7b50', time: '2018-02-07T22:42:28.403Z'
System info: host: 'WRIT-A05', ip: '172.16.7.127', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_111'
Driver info: driver.version: unknown
	at org.openqa.selenium.remote.server.ServicedSession$Factory.lambda$get$0(ServicedSession.java:128)
	at org.openqa.selenium.remote.server.ServicedSession$Factory.apply(ServicedSession.java:145)
	at org.openqa.selenium.remote.server.ActiveSessionFactory.lambda$apply$12(ActiveSessionFactory.java:176)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$11$1.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Unknown Source)
	at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
	at org.openqa.selenium.remote.server.ActiveSessionFactory.apply(ActiveSessionFactory.java:179)
	at org.openqa.selenium.remote.server.NewSessionPipeline.lambda$null$2(NewSessionPipeline.java:65)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.util.Collections$2.tryAdvance(Unknown Source)
	at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
	at org.openqa.selenium.remote.server.NewSessionPipeline.lambda$createNewSession$3(NewSessionPipeline.java:68)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.stream.DistinctOps$1$2.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.util.stream.Streams$StreamBuilderImpl.tryAdvance(Unknown Source)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Unknown Source)
	at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
	at org.openqa.selenium.remote.server.NewSessionPipeline.createNewSession(NewSessionPipeline.java:71)
	at org.openqa.selenium.remote.server.commandhandler.BeginSession.execute(BeginSession.java:65)
	at org.openqa.selenium.remote.server.WebDriverServlet.lambda$handle$0(WebDriverServlet.java:242)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

**Comment by MFaizanChughtai** [comment]
Please help me resolve this issue. I have latest Selenium server. i.e. 3.14.0 updated chrome driver but still system is unable to create a new session and shows the above error

**Comment by barancev** [comment]
`Build info: version: '3.9.1', revision: '63f7b50', time: '2018-02-07T22:42:28.403Z'`
Please upgrade.

**Comment by MFaizanChughtai** [comment]
I have tried this with downgrade selenium 3.9.1 but that does not work as well. I tried 3.14 as well. still same results

**Comment by barancev** [comment]
This is a question rather than an issue. Please send questions to the [selenium user group](https://groups.google.com/forum/#!forum/selenium-users)

**Comment by MFaizanChughtai** [comment]
R u serious ? Thats n issue brother. If u are unable to resolve it, be a man n admit it

**Comment by barancev** [comment]
Sorry for bad wording. It must be a *configuration* issue but not a bug in selenium code. Selenium is supported by community. The main support channel is the mail list, but not the issue tracker.

Most probably you don't have `chromedriver' in PATH, reading selenium server log will give you more information.

**Comment by MFaizanChughtai** [comment]
![09-12-18 9-01-17 am](https://user-images.githubusercontent.com/43163844/45401546-84774e00-b66a-11e8-9d97-5479ad282796.png)

**Comment by MFaizanChughtai** [comment]
is this helps finding out the issue?

**Comment by MFaizanChughtai** [comment]
![09-12-18 10-56-47 am](https://user-images.githubusercontent.com/43163844/45405007-a8db2680-b67a-11e8-9379-1ebc7f4d2f03.png)

**Comment by barancev** [comment]
Please use [selenium user group](https://groups.google.com/forum/#!forum/selenium-users) as the support channel.

**Comment by gopinath386** [comment]
Unable to create new service: ChromeDriverService Error i have also faced. For IEdriver also error to create new service.
I have used 3 node 1 hub. I set hub also one node. From hub machine i am executing the code. Selenium webdriver parallel execution. 
I have set the drivers in system environment also in selenium code with driver path.

Issue: From hub system drivers not start to execute the code. Error message "org.openqa.selenium.SessionNotCreatedException: Unable to create new service: ChromeDriverService"
Refer the attached screen shot too.
![error_run](https://user-images.githubusercontent.com/43443459/45821969-4ea41c00-bc9f-11e8-98bf-8c0e3185a837.jpg)
![nodes](https://user-images.githubusercontent.com/43443459/45821970-4ea41c00-bc9f-11e8-824a-ffe752dd189e.jpg)

Details:
1. Selenium java 3.14, Standalone 3.14 jar, TestNG 6.14.2,   Java 1.8.0_144 and chromedriver 2.42. 
2. Chrome as 69.0.3497.100.
3. Hub & node 1 for System as Windows Server 2008 R2 Enterprise, 64-bit. other two nodes System as Windows Server 2012 R2 Standard, 64-bit and all other is same properties.

Note: Same code working for other 2 nodes. Same chromedrivers only used in all three nodes.

Kindly check and advice. what is the root cause for the issue. Why the error shows only for the hub machine.
